wordle-web/scripts/wordOptions.ts

```diff
@@ -0,0 +1,37 @@
+import { WordsService } from '~/scripts/wordsService'
+
+ export class wordOptions {
+   words: string[] = []
+   count: number = 0
+
+   validWords(word: string): string[] {
+     this.words = []
+     if (!word.includes('?')) {
+       this.words.push('')
+       this.count = 0
+       return this.words
+     }
+
+     let expression: string = '^[A-Z]{0,5}'
+     for (let i = 0; i < 5; i++) {
+       if (word.charAt(i) === '?') {
+         expression += '[A-Z]'
+       } else {
+         expression += word.charAt(i)
+       }
+     }
```

Comment:
Expression should be empty at the beginning of this loop.

I was able to get the hint button to display the proper strings by replacing the above code with the following...

```
let expression: string = ''
     let expression: string = ''
     for (let i = 0; i < 5; i++) {
       if (word.charAt(i) === '?') {
         expression += '.'
       } else {
         expression += word.charAt(i)
       }
     }
     }
```

However, some tests were still not passing...